server/src/main/java/org/opensearch/index/translog/TranslogManager.java

```diff
@@ -33,6 +33,12 @@ public interface TranslogManager {
      */
     int recoverFromTranslog(TranslogRecoveryRunner translogRecoveryRunner, long localCheckpoint, long recoverUpToSeqNo) throws IOException;
 
+    /**
+     * Creates a new history snapshot from the translog file instead of the lucene index.
+     */
+    Translog.Snapshot newChangesSnapshot(long fromSeqNo, long toSeqNo, boolean requiredFullRange)
+        throws IOException,UnsupportedOperationException;
+
```

Comment:
`UnsupportedOperationException ` isn't a checked exception?

Reply:
Its a runtime exception: https://docs.oracle.com/javase/7/docs/api/java/lang/UnsupportedOperationException.html

Reply:
so that shouldn't necessarily be a part of the method signature and forced to be handled?

Reply:
Correct. Technically it isn't required to be part of method signature. I've kept it explicitly just to make the intention clear that this method isn't always guaranteed to return the operations. 
Let me know If you think i still should remove it from the signature.